src/app/modules/ix-table/components/ix-table-body/ix-table-body.component.ts

```diff
@@ -79,8 +80,30 @@ export class IxTableBodyComponent<T> implements AfterViewInit {
       && (this.dataProvider?.expandedRow?.[this.detailsRowIdentifier] === row?.[this.detailsRowIdentifier]);
   }
 
-  // TODO: Come up with a proper identifier
-  protected trackByIdentity<X>(item: X): X {
-    return item;
+  protected trackRowByIdentity(item: T): string {
+    return this.getTestAttr(item);
+  }
+
+  protected displayedColumnsForRow(row: T, columns: Column<T, ColumnComponent<T>>[]): Column<T, ColumnComponent<T>>[] {
+    for (const column of columns) {
+      column.setRow(row);
+    }
+    return columns;
+  }
+
+  protected trackColumnByIdentity(column: Column<T, ColumnComponent<T>>, index: number): string {
+    const trackById = this.getTestAttr(column.getRow());
+
+    if (column.title) {
+      return trackById + '-' + column.title + '-' + index;
+    }
+    const actionsColumn = column as Column<T, IxCellActionsComponent<T>>;
+    if (actionsColumn.actions?.length) {
```

Comment:
This makes `TableBodyComponent` aware of types of columns and their inner workings. 
This would only be useful if user changes visible columns and we don't want to re-render some of the cells, but addition of index reduces usefulness of it.

I'm fine with letting it keep tracking the object like it used to. Otherwise new system is needed, probably one where each ColumnComponent provides its own column id.

Also `trackSomethingByIdentity` needs to be renamed, because the word `identity` used to refer to tracking the object itself.